server/app/controllers/events_controller.rb

```diff
@@ -128,6 +128,18 @@ def get_users
   # POST /events/1/signup/1
   def signup
     event = Event.find_by_id(params[:id])
+    if event.registration_start_time and Time.now < event.registration_start_time
+      render json: {
+        errors: ["Registration has not yet open."],
```

Comment:
Grammatical mistake right here in the error returned from trying to pre-register, but that can easily be changed later.